interfaces/builtin/cups.go

```diff
@@ -0,0 +1,43 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import "github.com/ubuntu-core/snappy/interfaces"
+
+const cupsConnectedPlugAppArmor = `
+# Description: Can access cups daemon.
+# Usage: reserved
+#include <abstractions/cups-client>
```

Comment:
This policy looks ok but note that on Ubuntu the default user is in the lpadmin group. Currently this policy does not allow access to the org.opensuse.CupsPkHelper.Mechanism dbus interface (which is good), but a user in the admin group can access /run/cups/cups.sock and do things, such as adjust /etc/cups/cupsd.conf:

`
$ sudo grep Browsing /etc/cups/cupsd.conf # file perms need root
Browsing Off

$ cupsctl Browsing=On  # access to the socket if you are in the lpadmin group does not

$ sudo grep Browsing /etc/cups/cupsd.conf # file perms need root
Browsing ON
`

If you are not in the lpadmin group, you can not do this:
`
sudo -H -u clickpkg cupsctl Browsing=On
cupsctl: Forbidden
`

Note: running the above will remove comments from /etc/cups/cupsd.conf even if you revert with Browsing=off.

As such, this interface should probably not be autoconnected.

@tyhicks - I wonder if the launcher should drop supplemental groups when launching commends. Granted, that doesn't help for a malicious service that is running as root....